flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -1746,4 +1748,63 @@ PartitionReleaseStrategy getPartitionReleaseStrategy() {
 	ExecutionDeploymentListener getExecutionDeploymentListener() {
 		return executionDeploymentListener;
 	}
+
+	/**
+	 * Returns the mapping from intermediate result id to the ClusterPartitionDescriptor of its partitions
+	 *
+	 * @return The mapping from intermediate result id to the ClusterPartitionDescriptor of its partitions
+	 * 		   null, if the job has not yet finished
+	 */
+	public Map<IntermediateDataSetID, PersistedIntermediateResultDescriptor> getPersistedIntermediateResult() {
+		if (!JobStatus.FINISHED.equals(state)) {
+			return null;
+		}
+
+		final List<IntermediateDataSetID> persistedIntermediateDataSetIds = getPersistedIntermediateDataSetID();
```

Comment:
The logic here seems a little over complicated. Would iterating over the entrySet of the IntermediateResult and  handle the Id and partitions array in another method simpler?